### DISCUSSION TITLE
Home row mods and rolls

### DISCUSSION THREAD

**Discussion by jweickm** [precondition/precondition.github.io#22] [Q&A] [answered]
@precondition: I want to start this discussion by saying Thank you for such an amazing post. It has become the 'de facto' source to get started on home row mods, and also got me hooked on home row mods! 
I have a question regarding your article, where you write the following

```
case RMOD_T(r_letter):
        if (record->event.pressed && record->tap.count > 0) {
            if (get_mods() & MOD_BIT(KC_RSHIFT)) {
                unregister_mods(MOD_BIT(KC_RSHIFT));
                tap_code(r_s_letter);
                tap_code(r_letter);
                add_mods(MOD_BIT(KC_RSHIFT));
                return false;
            }
        }
        return true;
```
> Sticking with the Colemak example, imagine you have repetitive accidental mod activations when doing `RGUI_T(KC_O)+KC_U` when rolling keys for the bigram “ou”. You can add a case in your process_record_user for `KC_U` where you check the mod state for `RGUI` and override the default behaviour by sending “ou” instead of `GUI+U`.

Now, when I tried this out with `RGUI_T`, I ran into the problem that when I `unregister_mods(MOD_BIT(KC_RGUI));` or even `del_mods(MOD_BIT(KC_RGUI));` a tap of `KC_RGUI` is still sent (because the key down event was already registered) and windows will open the start menu or send a win-command. As you specifically mentioned the case with `RGUI` and `KC_U`, I was wondering if you got it to work in the way you described above?

Would love to get this to work, as accidental `GUI` keypresses are arguably the worst of the bunch^^

Many thanks in advance, 
jweickm

**Comment by precondition** [comment, marked as answer]
It's worked for me because my desktop environment (XFCE) doesn't bind any action to taps of the Windows key[1].

There are no band-aids to this approach that can mitigate the flashing modifiers issue. Your best bets are something along the lines of manna-harbour's [Bilateral Combinations](https://github.com/manna-harbour/qmk_firmware/blob/bilateral-combinations/docs/tap_hold.md#bilateral-combinations) or ZMK's [positional hold-taps](https://zmk.dev/docs/behaviors/hold-tap/#positional-hold-tap-and-hold-trigger-key-positions).

[1]:  I used to be bothered by the lack of this feature (open menu on GUI tap) when I was first switching from Windows to Linux but this frustration turned into gratefulness as I delved deeper in advanced tap-hold keys since this features turns out to cause many little annoyances and complications

**Reply by jweickm** [reply]
Hey,
Thank you for your quick reply. I see, that changes things a little bit. I have also seen the other implementations — especially the one by manna-harbour, but since I am not that experienced with C, I couldn't get it to work just yet. I have been using Windows Subsystem for Linux and thought that it is really great, but maybe there are some unexpected advantages to going full Linux after all^^
